themes/bootstrap5/js/covers.js

```diff
@@ -1,75 +1,111 @@
-        } else {
-          var span = $('<span class="cover-source-text"' + response.data.backlink_text + '</span>');
-          source.append(span);
-        }
-        var backlink_locations = response.data.backlink_locations;
-        if (backlink_locations.indexOf(context) >= 0) {
-          if (inlink === true) {
-            medium.append(source);
+VuFind.register('covers', function covers() {
+  function loadCoverByElement(data, element) {
+    var img = element.querySelector('img');
+    var spinner = element.querySelector('div.spinner');
+    var container = element.querySelector('div.cover-container');
+    var source = document.createElement('p');
+    source.classList.add('cover-source');
+    source.innerText = VuFind.translate('cover_source_label');
+    var context = data.context;
+
+    function coverCallback(response) {
+      if (typeof response.data.url !== 'undefined' && response.data.url !== false) {
+        img.src = response.data.url;
+        var inlink = element.parentElement.matches('a.record-cover-link');
+        var medium = img.closest('.media-left, .media-right, .carousel-item');
+        if (typeof response.data.backlink_text !== 'undefined') {
+          if (typeof response.data.backlink_url !== 'undefined') {
+            var link = document.createElement('a');
+            link.href = response.data.backlink_url;
+            link.classList.add('cover-backlink');
+            link.target = "_blank";
+            link.innerText = response.data.backlink_text;
+            source.appendChild(link);
           } else {
-            container.append(source);
+            var span = document.createElement('span');
+            span.classList.add('cover-source-text');
+            span.innerText = response.data.backlink_text;
+            source.appendChild(span);
+          }
+          var backlink_locations = response.data.backlink_locations;
+          if (backlink_locations.indexOf(context) >= 0) {
+            if (inlink === true) {
+              medium.appendChild(source);
+            } else {
+              container.appendChild(source);
+            }
           }
         }
-      }
-      if (inlink === true) {
-        img.detach();
-        medium.children('a').prepend(img);
-        container.parents('.ajaxcover').remove();
-      }
-    } else {
-      img.remove();
-      source.remove();
-      if (typeof response.data.html !== 'undefined') {
-        container.html(VuFind.updateCspNonce(response.data.html));
+        if (inlink === true) {
+          img.parentNode.removeChild(img);
+          var mediumLink = medium.querySelector('a');
+          mediumLink.parentNode.insertBefore(img, mediumLink);
+          container.closest('.ajaxcover').remove();
+        }
       } else {
-        container.html('');
+        img.remove();
+        source.remove();
+        if (typeof response.data.html !== 'undefined') {
+          container.innerHTML = VuFind.updateCspNonce(response.data.html);
```

Comment:
One other thing to consider: have you looked at #4065, which eliminates use of innerHTML? Maybe this can be adjusted to match the strategies found there.

Reply:
Makes sense. Done